jmxtrans-core/src/main/java/com/googlecode/jmxtrans/cluster/ClusterConfigClient.java

```diff
@@ -0,0 +1,225 @@
+package com.googlecode.jmxtrans.cluster;
+
+import org.apache.commons.configuration.Configuration;
+import org.apache.commons.configuration.ConfigurationException;
+import org.apache.commons.configuration.PropertiesConfiguration;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.CuratorFrameworkFactory;
+import org.apache.curator.framework.recipes.cache.PathChildrenCache;
+import org.apache.curator.framework.recipes.cache.PathChildrenCacheEvent;
+import org.apache.curator.framework.recipes.cache.PathChildrenCacheListener;
+import org.apache.curator.retry.ExponentialBackoffRetry;
+import org.apache.curator.utils.ZKPaths;
+import org.apache.zookeeper.CreateMode;
+import org.apache.zookeeper.data.Stat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * The main class that menages the connection to a cluster coordination
+ * service. If the cluster connection is enabled, the configuration will
+ * be queried from the cluster coordination service.
+ */
+public class ClusterConfigClient extends Thread implements ClusterConfigChangeListener{
+    private Configuration clConfiguration;
+    private String clServiceConnectionString;
+    private CuratorFramework clClient;
+    private String workerAlias = "";
+    private String configPath = "";
+    private String heartbeatPath = "";
+    PathChildrenCache jvmRootCache;
+    PathChildrenCache workerRootCache;
+    private Map<String, ClusterJvmConfigHandler> jvmConfigs;
+
+    private static final Logger log = LoggerFactory.getLogger(ClusterConfigClient.class);
+
+    public ClusterConfigClient(){
+        jvmConfigs = new HashMap<>();
+    }
+
+    public void initialize(File clConfigFile) throws ConfigurationException{
```

Comment:
This initialization logic should probably be moved to another class. That way you could ensure immutability of the ClusterConfigClient and ensure it is not possible to create an uninitialized config.